hack/tools/e2e-log-sync/README.md

```diff
@@ -0,0 +1,47 @@
+
+# E2e log sync
+
+## Prerequisites
+
+Start the Tilt development environment via `tilt up`.
+
+*Notes*:
+* If you only want to see imported logs, disable promtail.
+* If you want to drop all logs from Loki, just delete the Loki Pod in the `observability` namespace.
+
+## Import logs
+
+### Via Tilt UI
+
+There is now a `Import Logs` button on the top right on the [Loki resource page](http://localhost:10350/r/loki/overview).
+Just click on the downwards arrow, enter the URL of your ProwJob and click on `Import Logs`.
+This will download the logs from GCS and import them into Loki.
+
+### Via CLI
+
+Imports logs into Loki:
+```bash
+# Via ProwJob URL
+go run ./hack/tools/e2e-log-sync --log-prow-job-url=https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/6150/pull-cluster-api-e2e-main/1496233475802402816
+# Via bucket and log folder
+go run ./hack/tools/e2e-log-sync --log-bucket=kubernetes-jenkins --log-folder=pr-logs/pull/kubernetes-sigs_cluster-api/6150/pull-cluster-api-e2e-main/1496233475802402816
+```
+
+## View logs
+
+Now the logs are available:
+* via Grafana: `http://localhost:3001/explore`
+* via the Loki `logcli`:
+  ```bash
+  logcli query '{app="capd-controller-manager"}' --timezone=UTC --from="2022-02-22T10:00:00Z"
+  ```
+
+## Caveats
+
+* Make sure you query the correct time range.
+* The logs are currently uploaded with now as timestamp, because otherwise it would 
+  take a few minutes until the logs show up in Loki.
+
+## Future work
+
+* Make it possible to import logs from a local folder instead of GCS.
```

Comment:
Should I create a good first issue for this?

It should be relatively straightforward, we just have to read from filesystem instead of GCS

Reply:
Will include it in: https://github.com/kubernetes-sigs/cluster-api/pull/6189#issuecomment-1050079789

Did some simplifications around the flags too